lib/utils/forkedblockchain.js

```diff
@@ -234,52 +216,46 @@ ForkedBlockchain.prototype.getFallbackBlock = function(numberOrHash, cb) {
 };
 
 ForkedBlockchain.prototype.getBlock = function(number, callback) {
-  var self = this;
-
-  this.isFallbackBlockHash(number, function(err, isFallbackBlockHash) {
+  let checkFn;
+  const isBlockHash = this.isBlockHash(number);
+  if (isBlockHash) {
+    checkFn = this.isFallbackBlockHash;
+  } else {
+    checkFn = this.isFallbackBlock;
+  }
+  checkFn.call(this, number, (err, isFallback) => {
     if (err) {
       return callback(err);
     }
-    if (isFallbackBlockHash) {
-      return self.getFallbackBlock(number, callback);
+    if (isFallback) {
+      return this.getFallbackBlock(number, callback);
     }
 
-    self.isFallbackBlock(number, function(err, isFallbackBlock) {
-      if (err) {
-        return callback(err);
-      }
-
-      if (isFallbackBlock) {
-        return self.getFallbackBlock(number, callback);
-      }
-
-      // If we don't have string-based a block hash, turn what we do have into a number
-      // before sending it to getBlock.
-      function getBlockReference(value, callback) {
-        if (!self.isBlockHash(value)) {
-          self.getRelativeBlockNumber(value, callback);
-        } else {
-          callback(null, value);
-        }
-      }
-
-      getBlockReference(number, function(err, blockReference) {
+    const getBlock = BlockchainDouble.prototype.getBlock.bind(this);
+    if (isBlockHash) {
+      getBlock(number, callback);
+    } else {
+      this.getRelativeBlockNumber(number, (err, number) => {
         if (err) {
           return callback(err);
         }
-
-        BlockchainDouble.prototype.getBlock.call(self, blockReference, callback);
+        getBlock(number, callback);
       });
-    });
+    }
   });
 };
 
 ForkedBlockchain.prototype.getStorage = function(address, key, number, callback) {
-  this.lookupStorageTrie(address, function(err, trie) {
+  this.getLookupStorageTrie(this.stateTrie)(address, (err, trie) => {
```

Comment:
Reason for change: Make sure we use the new and improved storage trie instead of the old and busted one (which has been removed)